includes/promoted-jobs/class-wp-job-manager-promoted-jobs-status-handler.php

```diff
@@ -73,6 +68,33 @@ public function fetch_updates() {
 		}
 	}
 
+	/**
+	 * Checks if the update logic should be executed or not.
+	 *
+	 * @param int $current_time The current time.
+	 * @return bool True if the update logic should be executed, false otherwise.
+	 */
+	private function should_update( $current_time ) {
+		$interval = $this->get_current_interval();
+		if ( ! $interval ) {
+			// If the interval is not set or is zero, we don't update.
+			return false;
+		}
+		$last_execution = get_option( self::LAST_EXECUTION_OPTION_KEY, 0 );
+		return $current_time - $last_execution >= $interval;
+	}
+
+	/**
+	 * Get the current interval for fetching updates. This is either the interval for the cron or the webhook, depending
+	 * on the context of the current request.
+	 *
+	 * @return int The current interval for fetching updates.
+	 */
+	private function get_current_interval() {
+		$option_name = wp_doing_cron() ? self::CRON_INTERVAL_OPTION_KEY : self::WEBHOOK_INTERVAL_OPTION_KEY;
```

Comment:
I'm using `wp_doing_cron()` here to detect whether this is a CRON request or not and use the appropriate option key.